cpp/src/cider/exec/processor/DefaultBatchProcessor.cpp

```diff
@@ -123,7 +123,7 @@ void DefaultBatchProcessor::feedHashBuildTable(
     const std::shared_ptr<JoinHashTable>& hashTable) {
   // switch state from waiting to running once hashTable is ready
   this->state_ = BatchProcessorState::kRunning;
-  // TODO: feed the hashTable into nextGen context
+  this->codegen_context_->setHashTable(*hashTable);
```

Comment:
is it safe?

Reply:
Shared_ptr and unique_ptr can not be used in codegen context. So this can only pass raw pointer

Reply:
operator* returns a reference. https://en.cppreference.com/w/cpp/memory/shared_ptr/operator*
try `hashTable->get()`?

Reply:
already updated